src/tracing.rs

```diff
@@ -26,17 +33,53 @@ impl Feature for Tracing {
             .with_collector_endpoint(&config.tracing.opentelemetry_endpoint)
             .with_service_name(module_path!())
             .install_batch(opentelemetry::runtime::Tokio)?;
-
         let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
 
+        // tracing_subscriber lib currently does not support dynamically adding layer to registry
+        // accordingly to some condition. this can be verified in the following issues:
+        // https://github.com/tokio-rs/tracing/issues/575
+        // https://github.com/tokio-rs/tracing/issues/1708
+        //
+        // but there is a workaround described here:
+        // https://github.com/tokio-rs/tracing/issues/894
+        //
+        // the workaround consists of passing a optional of Layer to every conditional layer,
+        // so if Some(layer) is passed, that layer is active, if None the layer is inactive.
+        let (layer_format_json, layer_format_pretty, layer_format_hierarchical) =
+            match config.tracing.format {
+                TracingFormat::None => (None, None, None),
+                TracingFormat::Json => (Some(Layer::default().json()), None, None),
+                TracingFormat::Pretty => (
```

Comment:
So, reading a little bit on it, it doesn't seem like we actually need something like this, as we are not changing the registry again in runtime, we only set up it once. Maybe we could make `TracingFormat` return its own layer, and then just slap it into the registry?

Of course, the types are different, but you can return [impl Layer<S>](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html), and `Registry` should accept it.